packages/google-compute-engine-oslogin/utils/oslogin_utils.cc

```diff
@@ -470,6 +469,90 @@ bool ParseJsonToPasswd(const string& json, struct passwd* result,
   return ValidatePasswd(result, buf, errnop);
 }
 
+bool ParseJsonToGroup(const string& json, struct group* result,
+                       BufferManager* buf, int* errnop) {
+  json_object* root = NULL;
+  root = json_tokener_parse(json.c_str());
+
+  if (root == NULL) {
+    *errnop = ENOENT;
+    return false;
+  }
+
+  json_object* groups = NULL;
+  if (!json_object_object_get_ex(root, "posixGroups", &groups)) {
+    // TODO: set errnop
```

Comment:
Following the pattern from the other methods, would EINVAL make sense here (and at other error paths)?